WcaOnRails/app/models/competition.rb

```diff
@@ -509,12 +524,20 @@ def end_date
   end
 
   private def events_must_be_valid
-    invalid_events = events - Event.all_official - Event.all_deprecated
+    invalid_events = events - Event.official - Event.deprecated
```

Comment:
Same question... are we possibly looking at events that have been marked for destruction here?